103-bundle-runtime.md

```diff
@@ -241,15 +235,9 @@ The `/cnab/app/image-map.json` file mounted in the invocation image will be:
     "my-microservice": {
         "description": "my microservice",
         "digest": "sha256:aaaaaaaaaaaa...",
-        "image": "technosophos/microservice:1.2.3",
-        "refs": [
-            {
-                "field": "image.1.field",
-                "path": "image1path"
-            }
-        ]
+        "image": "technosophos/microservice:1.2.3"
     }
 }
 ```
 
-The run tool MAY use this file to modify its behavior, if declarative substitution is not enough.
\ No newline at end of file
+The run tool MAY use this file to modify its behavior. For example, a run tool could choose to copy an image from one registry to another, and accordingly adjust the locations used by the installation logic.
```

Comment:
This seems a bit far fetched. A more realistic scenario is:
```
For example, a CNAB publishing tool can copy a CNAB and all its component images in a new repository without having to rebuild the invocation image: it just have to modify the CNAB's image map. At invocation time, the run tool will use the injected image map to update the image references for the impacted components.
```

Reply:
@simonferquel maybe I am misunderstanding what you believe to be far-fetched, but I believe the scenario that is described is precisely what would occur for many bundle consumers running bundles inside of a network-partitioned environment. In this case, a CNAB publishing tool would have no understanding of the registry inside of the physical barrier. Instead, the images would need to be relocated from a DMZ into a registry inside the air-gapped network such that the platform running inside that network could take advantage of them.

Reply:
Yes, but this should not be the role of the invocation image to do so (the run tool). 
It should be an internal provisioning tool run within the DMZ (e.g., using cnab-to-oci to pull a bundle from a public registry, and push it back to an internal one).

This is a scenario we already cover with cnab-to-oci and docker-app: 
- docker-app creates bundles with a /run/tool that understands the injected image-map
- cnab-to-oci can push a bundle with all its images in a single repository, rewriting the image-map so that component images are replaced with digested references in this repository
- cnab-to-oci can then be used to re-push the same bundle to different registries/repositories, each time rewriting the image-map.
- when docker-app (or duffle) is used to install such a rewritten bundle, it injects the image-map with the new images reference. 

The run tool within the invocation image never copies anything from a registry to another, it just consumes the image-map which might point to different image references than those defined at build time.